src/test/java/org/junit/internal/MethodSorterTest.java

```diff
@@ -147,6 +148,21 @@ void epsilon() {
     public void testJvmMethodSorter() {
         Method[] fromJvmWithSynthetics = DummySortJvm.class.getDeclaredMethods();
         Method[] sorted = MethodSorter.getDeclaredMethods(DummySortJvm.class);
+
+        Comparator<Method> methodComparator = new Comparator<Method>() {
+            @Override
+            public int compare(Method m1, Method m2) {
+                int nameComparison = m1.getName().compareTo(m2.getName());
+                if (nameComparison != 0) {
+                    return nameComparison;
+                }
+                return m1.toString().compareTo(m2.toString());
+            }
+        };
+        
+        Arrays.sort(fromJvmWithSynthetics, methodComparator);
+        Arrays.sort(sorted, methodComparator);
```

Comment:
This test now simply verifies that the two methods return the same values, ignoring order, so with these changes it's not a particularly good test.

The ordering of methods returned by `getDeclaredMethods()` is no longer guaranteed as of JDK 7. I think the only way to make this not flaky would be to skip the test if the JDK is 7 or greater.